src/brpc/couchbase.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <couchbase/cluster.hxx>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace brpc {
+
+class CouchbaseWrapper {
+ public:
+  CouchbaseWrapper() = default;
+  ~CouchbaseWrapper() { CloseCouchbase(); }
+  // Initialize Couchbase connection (call once at startup)
+  bool InitCouchbase(const std::string& connection_string,
+                     const std::string& username, const std::string& password);
+
+  // Get document by key
+  std::pair<bool, std::string> CouchbaseGet(
+      const std::string& key, const std::string& bucket_name,
+      const std::string& scope = "_default",
+      const std::string& collection = "_default");
+
+  // Upsert (insert/update) document
+  bool CouchbaseUpsert(const std::string& key, const std::string& value,
+                       const std::string& bucket_name,
```

Comment:
We may want to file an improvement issue for the future to use fully qualified buckets/scopes/collections.  This offers protection from a delete-recreate of a collection being logically different.  I think this is fine for now though.

Reply:
I do mention it as it may change the API in the future.  I guess we could add something that isn't a string.